### PR TITLE
chore: Upgrades Atlas SDK to v20231115012

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): doing as part of CLOUDP-228077

This new SDK version includes the new `integrationId` property for alert configuration notifications.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/alertconfiguration/model_alert_configuration.go [hunk 1]
```diff
@@ -81,13 +81,13 @@ func NewMetricThreshold(tfMetricThresholdConfigSlice []TfMetricThresholdConfigMo
...
+	matchers := make([]admin.StreamsMatcher, len(list))
 	for i, matcher := range list {
-		matchers[i] = map[string]any{
-			"fieldName": matcher.FieldName.ValueString(),
-			"operator":  matcher.Operator.ValueString(),
-			"value":     matcher.Value.ValueString(),
+		matchers[i] = admin.StreamsMatcher{
+			FieldName: matcher.FieldName.ValueString(),
+			Operator:  matcher.Operator.ValueString(),
+			Value:     matcher.Value.ValueString(),
```
Comment (AgustinBettati) [inline internal/service/alertconfiguration/model_alert_configuration.go:90]: Only change was in alert configuration matchers model which is now typed (was a `map[string]any` before).

File: internal/service/alertconfiguration/model_alert_configuration_test.go [hunk 2]
```diff
@@ -208,9 +202,7 @@ func TestMatcherSDKToTFModel(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			resultModel := alertconfiguration.NewTFMatcherModelList(tc.SDKResp, tc.currentStateMatcher)
-			if !reflect.DeepEqual(resultModel, tc.expectedTFModel) {
-				t.Errorf("created terraform model did not match expected output.\n Expected: %s \n Result: %s", tc.expectedTFModel, resultModel)
-			}
+			assert.Equal(t, tc.expectedTFModel, resultModel, "created terraform model did not match expected output")
```
Comment (AgustinBettati) [inline internal/service/alertconfiguration/model_alert_configuration_test.go:205]: small pass by refactor

**Review by EspenAlbert — APPROVED** [review]
LGTM!

**Review by maastha — APPROVED** [review]
LGTM

**Pull request merged**